### PR TITLE
Initial atempts to fix bug with current screen being indicator status

### PR DESCRIPTION
**Pull request by Yaraslaut**
Closes https://github.com/contour-terminal/contour/issues/1222

### CODE REVIEW

File: src/vtbackend/Terminal.cpp [hunk 1]
```diff
@@ -831,8 +831,8 @@ void Terminal::sendMouseMoveEvent(Modifier modifier,
     }
 
     auto const shouldExtendSelection = shouldExtendSelectionByMouse(newPosition, pixelPosition);
-
     auto relativePos = _viewport.translateScreenToGridCoordinate(newPosition);
+    fmt::print(" shouldExtend {}, relativePos {} \n", shouldExtendSelection, relativePos);
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:835]:
```suggestion
```

File: src/vtbackend/Terminal.cpp [hunk 2]
```diff
@@ -852,18 +852,28 @@ void Terminal::sendMouseMoveEvent(Modifier modifier,
     if (_leftMouseButtonPressed)
     {
         if (!selectionAvailable())
+        {
+            fmt::print("selection not available\n");
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:856]:
```suggestion
```

File: src/vtbackend/Terminal.cpp [hunk 3]
```diff
@@ -852,18 +852,28 @@ void Terminal::sendMouseMoveEvent(Modifier modifier,
...
             setSelector(
                 std::make_unique<LinearSelection>(_selectionHelper, relativePos, selectionUpdatedHelper()));
+        }
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:859]:
```suggestion
```

File: src/vtbackend/Terminal.cpp [hunk 4]
```diff
@@ -852,18 +852,28 @@ void Terminal::sendMouseMoveEvent(Modifier modifier,
...
         {
             if (currentScreen().isCellEmpty(relativePos)
                 && !currentScreen().compareCellTextAt(relativePos, 0x20))
-                relativePos.column = ColumnOffset { 0 } + *(_settings.pageSize.columns - 1);
-            _state.viCommands.cursorPosition = relativePos;
+            {
+                relativePos.column = ColumnOffset { 0 } + unbox(_settings.pageSize.columns - 1);
+                fmt::print("change coloums {} \n", relativePos);
+                _state.viCommands.cursorPosition = relativePos;
+            }
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:868]:
```suggestion
            _state.viCommands.cursorPosition = relativePos;
```

File: src/vtbackend/Terminal.cpp [hunk 5]
```diff
@@ -852,18 +852,28 @@ void Terminal::sendMouseMoveEvent(Modifier modifier,
...
             if (_state.inputHandler.mode() != ViMode::Insert)
+            {
                 _state.inputHandler.setMode(selector()->viMode());
+            }
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:872]:
```suggestion
            _state.inputHandler.setMode(selector()->viMode());
```

File: src/vtbackend/Terminal.cpp [hunk 6]
```diff
@@ -852,18 +852,28 @@ void Terminal::sendMouseMoveEvent(Modifier modifier,
...
             if (selector()->extend(relativePos))
+            {
                 breakLoopAndRefreshRenderBuffer();
+            }
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:876]:
```suggestion
            breakLoopAndRefreshRenderBuffer();
```

File: src/vtbackend/Viewport.cpp [hunk 7]
```diff
@@ -12,22 +12,28 @@ namespace vtbackend
 
 bool Viewport::scrollUp(LineCount numLines)
 {
-    return scrollTo(
-        std::min(_scrollOffset + numLines.as<ScrollOffset>(), boxed_cast<ScrollOffset>(historyLineCount())));
+    auto offset =
+        std::min(_scrollOffset + numLines.as<ScrollOffset>(), boxed_cast<ScrollOffset>(historyLineCount()));
+    viewportLog()("{} = min({} + {}, {})", offset , _scrollOffset, numLines, historyLineCount());
```
Comment (christianparpart) [inline src/vtbackend/Viewport.cpp:17]:
```suggestion
```

**Review by christianparpart — COMMENTED** [review]
i think this should be more than one commit, with all the debug prints/logs removed :)

I can take this PR over also, so you can continue working on config management refactor

**Comment by Yaraslaut** [comment]
I updated this PR @christianparpart  and at the moment it fixes issue with screen selection and reproduce bug from master branch with same selection happening for `LineOffset(0)` and status line

**Pull request merged**